src/utils/Database/LSMTree/Generators.hs

```diff
@@ -574,6 +590,13 @@ instance Arbitrary SerialisedBlob where
   arbitrary = SerialisedBlob <$> genRawBytes
   shrink (SerialisedBlob rb) = SerialisedBlob <$> shrinkRawBytes rb
 
+newtype LargeRawBytes = LargeRawBytes RawBytes
+  deriving Show
+
+instance Arbitrary LargeRawBytes where
+  arbitrary = genRawBytesSized (4096*3) >>= fmap LargeRawBytes . genSlice
+  shrink (LargeRawBytes rb) = map LargeRawBytes (shrink rb)
```

Comment:
So `LargeRawBytes` is just a wrapper around `RawBytes`, so the right place to fix this would be in the `RawBytes` shrinker.

That one uses
```
  shrink rb = shrinkRawBytes rb ++ shrinkSlice rb
```
where `shrinkRawBytes` does the list thing and `shrinkSlice` uses a combo of take and drop.

So presumably we should just reverse the order of these two shrinkers and put the slice one first.

Reply:
The values generated by `Arbitrary RawBytes` should not be as large, but I agree, it's better to address the issue there.

Reversing the order is good, but doesn't entirely prevent the problem. If a property only fails for large (e.g. 4k+) `RawBytes`, once the shrinker reduced the length to that, it will iterate through all the list shrinks of a 4k+ elements list.